File: packages/eventindexer/contracts/erc721/abi.go
package erc721

var (
	ABI = `[
		{
			"constant":true,
			"inputs":[
				{
					"name":"_tokenId",
					"type":"uint256"
				}
			],
			"name":"tokenURI",
			"outputs":[
				{
					"name":"",
					"type":"string"
				}
			],
			"payable":false,
			"stateMutability":"view",
			"type":"function"
		},
		{
			"constant": true,
			"inputs": [],
			"name": "symbol",
			"outputs": [
				{
					"name": "",
					"type": "string"
				}
			],
			"payable": false,
			"stateMutability": "view",
			"type": "function"
		}
	]`
)

File: packages/usdc-faucet-ui/src/lib/abi.ts
export const faucetAbi = [
  {
    type: 'function',
    name: 'claim',
    stateMutability: 'nonpayable',
    inputs: [],
    outputs: [],
  },
  {
    type: 'function',
    name: 'nextClaimAt',
    stateMutability: 'view',
    inputs: [{ name: '_claimer', type: 'address' }],
    outputs: [{ name: 'nextClaimAt_', type: 'uint256' }],
  },
  {
    type: 'function',
    name: 'claimAmount',
    stateMutability: 'view',
    inputs: [],
    outputs: [{ name: 'claimAmount_', type: 'uint256' }],
  },
  {
    type: 'function',
    name: 'token',
    stateMutability: 'view',
    inputs: [],
    outputs: [{ name: 'token_', type: 'address' }],
  },
] as const;

export const erc20Abi = [
  {
    type: 'function',
    name: 'balanceOf',
    stateMutability: 'view',
    inputs: [{ name: 'account', type: 'address' }],
    outputs: [{ name: 'balance', type: 'uint256' }],
  },
  {
    type: 'function',
    name: 'decimals',
    stateMutability: 'view',
    inputs: [],
    outputs: [{ name: '', type: 'uint8' }],
  },
  {
    type: 'function',
    name: 'name',
    stateMutability: 'view',
    inputs: [],
    outputs: [{ name: '', type: 'string' }],
  },
  {
    type: 'function',
    name: 'symbol',
    stateMutability: 'view',
    inputs: [],
    outputs: [{ name: '', type: 'string' }],
  },
] as const;
